Fix the module wizard and section wizard scripts

m-vo opened the pull request:
Fixes #8129

This PR introduces templates for the `ModuleWizard` and `SectionWizard`, migrates the corresponding new-old vanilla JS to a Stimulus controller (`row-wizard`) and introduces template snapshots<sup>*)</sup> for each of the wizards rows.

*) The issue from #8129 shows a nasty problem: When clicking on "copy", we basically duplicated the `<tr>` element with all its children. This included things like choices instances that were already in their "wrapped" state (Choices modifies the DOM), so when inserting these, the respective controller connected, tried to init and failed (in the case of Choices, because the Choices script finds a certain data-attribute indicating an "active" instance). 

The new solution wraps the rows in `<template>` elements and lets the controller unwrap them. When doing so, a snapshot of the original HTML is created and stored. This snapshot is both used when copying a row and when restoring the original state before a cache entry is made.

m-vo committed:
refactor module wizard HTML into a template

m-vo committed:
migrate the module wizard script to a Stimulus controller and implement snapshotting

m-vo committed:
drop old script

m-vo committed:
build assets

m-vo committed:
build assets

m-vo committed:
simplify teardown

m-vo commented on core-bundle/assets/controllers/module-wizard-controller.js:
```diff
@@ -0,0 +1,147 @@
+        this._makeSortable();
+    }
+
+    rowTargetDisconnected() {
```
```suggestion
    rowTargetDisconnected(row) {
```

m-vo replied:
Assets need to be rebuilt.

zoglo replied:
Did that in 19f7e966d648532d02c7e4a0c4917f8c80359980 :)

m-vo committed:
Update core-bundle/assets/controllers/module-wizard-controller.js

zoglo committed:
Rebuild the assets

leofeyer commented on core-bundle/assets/controllers/module-wizard-controller.js:
```diff
@@ -0,0 +1,147 @@
+    delete(event) {
+        const row = this._getRow(event);
+
+        row.remove();
```
```suggestion
        this._getRow(event).remove();
```

leofeyer commented on core-bundle/assets/controllers/module-wizard-controller.js:
```diff
@@ -0,0 +1,147 @@
+    }
+
+    enable(event) {
+        event.target.previousElementSibling.checked ^= 1;
```
Is `^=` correct here?

zoglo replied:
Had to look it up as well, it's a bitwise XOR that toggles the checkbox:
true => false
false => true

leofeyer replied:
🦊

m-vo committed:
Update core-bundle/assets/controllers/module-wizard-controller.js

Co-authored-by: Leo Feyer <1192057+leofeyer@users.noreply.github.com>

m-vo committed:
abstract module wizard to row wizard

m-vo committed:
refactor SectionWizard and make use of the row wizard Stimulus controller

m-vo committed:
re-add handling for when trying to delete the last row

m-vo committed:
build assets

m-vo commented:
This PR now includes the same changes for the `SectionWizard´ - I updated the PR title and initial description.

As both the `ModuleWizard` and the `SectionWizard` behave in a very similar way, instead of duplicating a lot of code, I implemented a `row-wizard` Stimulus controller that is used in both.

leofeyer committed:
Merge branch '5.5' into bugfix/module-wizard

leofeyer committed:
Rebuild the assets

leofeyer reviewed (changes requested):
Unfortunately, the module wizard does not work in Safari:

https://github.com/user-attachments/assets/13f8e1d4-52e0-472c-abf6-76e621ca51d8

The section wizard runs fine. Both run fine in Firefox.

m-vo commented:
Time for some BrowserStack action. I'll try to fix this...

m-vo committed:
make sure Choice's cloned inputs are not used when syncing inputs

leofeyer commented:
Thank you @m-vo.
